File: TaskFullStackPt2/ClientApp/src/app/Task.ts
export interface Task{
    id: number;
    name: string;
    instructions: string;
    dueDate: Date;
    isCompleted: boolean;
}

export class Convert {
    public static toTask(json: string): Task {
        return JSON.parse(json);
    }
    public static taskToJson(value: Task): string {
        return JSON.stringify(value);
    }
}

File: TaskFullStackPt2/Models/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TaskFullStackPt2.Models
{
    public class Task
    {

        public int Id { get; set; }
        public string Name { get; set; }
        public string Instructions { get; set; }
        public string DueDate { get; set; }
        public bool IsCompleted { get; set; }

        public Task(int id, string name, string instructions, string dueDate, bool isCompleted)
        {
            Id = id;
            Name = name;
            Instructions = instructions;
            DueDate = dueDate;
            IsCompleted = isCompleted;
        }
        public Task()
        {

        }
    }
}
